stdlib/public/Concurrency/TaskStatus.cpp

```diff
@@ -309,15 +268,15 @@ bool swift::removeStatusRecord(TaskStatusRecord *record) {
 
   while (true) {
     // Wait for any active lock to be released.
-    if (oldStatus.isLocked())
+    if (oldStatus.isStatusRecordLocked())
       waitForStatusRecordUnlock(task, oldStatus);
 
     // If the record is the innermost record, try to just pop it off.
     if (oldStatus.getInnermostRecord() == record) {
       ActiveTaskStatus newStatus =
         oldStatus.withInnermostRecord(record->getParent());
       if (status.compare_exchange_weak(oldStatus, newStatus,
-             /*success*/ std::memory_order_release,
+             /*success*/ std::memory_order_relaxed,
```

Comment:
I don't see why this has to be a release barrier, it seems like

(a) we're simply updating the ActiveTaskStatus's TaskStatusRecord pointer with what was the previous head pointer and all the state we want to publish is within that atomic.

(b) We're not making any other writes to the new record we're publishing, such that it needs to happen before this newStatus is visible.

That's why I changed this from a release barrier to a relaxed. Let me know if I'm missing something here.

Reply:
Your analysis makes sense to me. Maybe this was written with the general idea of "we're publishing a pointer, better make sure the contents are published first" without realizing that the contents aren't being changed here.

Reply:
Yeah, I think this is fine.